content/ADR-145-pick-realm-v2.md

```diff
@@ -59,46 +79,137 @@ https://github.com/decentraland/kernel/blob/main/packages/shared/dao/index.ts#L6
 
-  if (currentChecks && result.accepting_users) {
+  if (currentChecks && result.acceptingUsers) {
     // ...
   }
 
   return undefined
 }
 ```
 
-_Configuration_:
+### Catalyst Version
+
+_Description_:
+
+Every Catalyst node provides information about the versions of its Content, Lambdas, BFF, and Comms components. The Kernel reads these version fields and compares them with the minimum required version specified in the rule configuration. As a result, the configuration has the flexibility to specify minimum required versions for one or more services, allowing the Catalyst to be considered as a candidate only if it meets these requirements. This means that the configuration file can specify minimum required versions for X and Xn services, giving the flexibility to specify versions for one or more services as needed.
+
 
-This configuration will be at BFF (Catalyst) level
+_Pseudo Code_:
 
 ```typescript
-export type AcceptingUsersConfig = {
-  config?: 
-          { maxAmountOfUsers: number;
-            maxCpu: number;
-            maxMemory: number;
-          }
+export function catalystVersionLink(configuration: CatalystVersionConfig) {
+  const minimumRequiredVersions = configuration
+
+  Object.keys(minimumRequiredVersions).forEach(([serviceName, serviceRequiredVersion]) => {
+    validCandidates = picked.filter(candidate => meetsVersion(candidate, serviceName, serviceRequiredVersion))
+    // ...
+  })
+
+  return validCandidates
 }
 ```
 
+_Configuration_:
 
-### Catalyst Version Link
+```typescript
+export type CatalystVersionConfig = {
+  config?: 
+          { comms?: string;
+            bff?: string;
+            content?: string;
+            lambdas?: string;
+          }
+}
+```
 
+### Force Catalyst
 
 _Description_:
 
-Each Catalyst (BFF) exposes the version of Content and Lambdas that it's using. The Kernel should read that field, compare them and prioritize the ones having the highest value.
-As Content Server and Lambdas share the same version for each Catalyst and they are using semver, there is always an unique way to sort the candidates.
-
+This rule will have multiple Catalysts, sorted by priority, defined in its configuration. The rule will attempt to connect to the first Catalyst in the list. If it is unavailable, it will continue trying with the subsequent Catalysts specified in the configuration in order of priority.
 
 _Pseudo Code_:
 
 ```typescript
-export function versionLink() {
-  const catalystsWithVersion: { realm: string, version: string } = peers.sortByVersion()
-  const versionBuckets: PriorityQueue<{ realm: string, version: string }[]> = groupByVersions(catalystsWithVersion)
-  // This selects randomly between the candidates in the selected bucket 
-  // And chooses the bucket with the same algorithm used for close peers
-  return selectFirstByScore(context, score, definitiveDecisionThreshold)
+export function catalystVersionLink(configuration: ForceCatalystConfig) {
```

Comment:
If possible I'd try to use a different analogy than the current algorithm "links" it has proven hard to reason and understand

Reply:
I'm replacing this link concept for rules.

Reply:
Done here -> [feedback: replace links concept by rules](https://github.com/decentraland/adr/pull/220/commits/8a31761d537647dc8168a52bbebb6572da603234)